src/luarocks/fs/lua.lua

```diff
@@ -544,86 +544,183 @@ end
 if socket_ok then
 
 local ltn12 = require("ltn12")
+local http = require("socket.http")
 local luasec_ok, https = pcall(require, "ssl.https")
 
-local redirect_protocols = {
-   http = http,
-   https = luasec_ok and https,
-}
+-- Patch ltn12 source used by LuaSocket to get fixed length
```

Comment:
My only concern is: is this safe/stable across LuaSocket versions?...

I also noticed I couldn't get the length when I wrote the `spinner` patch (or else I would have made a progress bar). I also noticed that, at least in my tests, the ltn12 `step` wasn't that responsive (that's why I ended up with just a generic "something's happening" spinner).